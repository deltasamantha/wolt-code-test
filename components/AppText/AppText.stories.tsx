import {Meta, StoryObj} from "@storybook/react-native";
import {View} from "react-native";

import AppText, {Props} from "./AppText";
import {storyStyles} from "./AppText.styles";

/**
 * Story of AppText
 */
const AppTextMeta: Meta<Props> = {
  title: "AppText",
  component: AppText,
  argTypes: {
    type: {
      description: "Text type of the text.",
      control: "radio",
      options: ["bold", "regular", "semiBold"],
    },
    color: {
      description: "Text color type of the text.",
      control: "radio",
      options: {light: "light", dark: "dark", accent: "accent"},
    },
    size: {
      description: "Text size type of the text.",
      control: "radio",
      options: ["s", "m", "l"],
    },
  },
  args: {
    children: "Code test",
  },
  decorators: [
    Story => (
      <View style={storyStyles.rootContainer}>
        <Story />
      </View>
    ),
  ],
};

export default AppTextMeta;

export const Primary: StoryObj<Props> = {
  args: {
    children: "Primary text",
  },
};

export const LightColorLargeBold: StoryObj<Props> = {
  args: {
    type: "bold",
    color: "light",
    size: "l",
  },
};
export const AccentColorLargeSemiBold: StoryObj<Props> = {
  args: {
    type: "semiBold",
    color: "accent",
    size: "l",
  },
};
