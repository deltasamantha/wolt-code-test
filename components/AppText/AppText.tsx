import React, {ReactNode} from "react";
import {Text, TextProps, StyleProp, TextStyle} from "react-native";

import {TextSizeType, TextType, TextColorType} from "../../models/TextType";
import {
  getFontColor,
  getFontFamily,
  getFontLineHeight,
  getFontSize,
} from "../../utils/TextUtils";

export interface Props extends TextProps {
  children?: ReactNode;
  size?: TextSizeType;
  type?: TextType;
  color?: TextColorType;
  style?: StyleProp<TextStyle>;
}

const AppText: React.FC<Props> = ({
  children,
  size = "s",
  type = "regular",
  color = "dark",
  style,
  ...textProps
}: Props) => {
  return (
    <Text
      style={[
        {
          fontSize: getFontSize(size),
          lineHeight: getFontLineHeight(size),
          fontFamily: getFontFamily(type),
          color: getFontColor(color),
        },
        style,
      ]}
      {...textProps}>
      {children}
    </Text>
  );
};

export default AppText;
