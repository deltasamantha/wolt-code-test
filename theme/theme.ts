const theme = {
  colors: {
    colorBackground: "#ABD6EA",
    colorPrimary: "",
    colorAccent: "#5BCB02",
    colorBlack: "#000000",
    colorWhite: "#FFFFFF",
    fontColorDark: "#202125",
    fontColorLight: "#A1A2A4",
  },
  borderRadius: {
    sm: 5,
    md: 10,
    lg: 20,
  },
  spacing: {
    xs: 2,
    sm: 4,
    md: 10,
    lg: 16,
    xl: 24,
  },
  fontSize: {
    small: 12,
    medium: 16,
    large: 24,
  },
  fontLineHeight: {
    small: 16,
    medium: 22,
    large: 30,
  },
};

export default theme;
