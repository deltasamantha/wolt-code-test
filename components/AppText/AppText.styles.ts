import {StyleSheet} from "react-native";

import theme from "../../theme/theme";

export const storyStyles = StyleSheet.create({
  rootContainer: {
    flex: 1,
    paddingHorizontal: theme.spacing.lg,
    backgroundColor: theme.colors.colorWhite,
    justifyContent: "center",
    alignItems: "center",
  },
});
