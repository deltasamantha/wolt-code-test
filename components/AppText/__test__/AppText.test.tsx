import {render, cleanup, screen} from "@testing-library/react-native";
import "@testing-library/jest-native/extend-expect";

import theme from "../../../theme/theme";
import AppText from "../AppText";

afterEach(cleanup);

describe("AppText", () => {
  it("Should render AppText properly", () => {
    const sampleText = "Code test";
    const {toJSON} = render(<AppText>{sampleText}</AppText>);
    const text = screen.getByText(sampleText);
    expect(text.props.children).toBeDefined();
    expect(text.props.children).toEqual(sampleText);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Should render dark color variant", () => {
    const sampleText = "Code test";
    const color = theme.colors.colorDark;
    render(<AppText color="dark">{sampleText}</AppText>);
    const text = screen.getByText(sampleText);
    expect(text).toBeDefined();
    expect(text).toHaveTextContent(sampleText);
    expect(text).toHaveStyle({
      color,
    });
  });

  it("Should render light color variant", () => {
    const sampleText = "Code test";
    const color = theme.colors.colorLight;
    render(<AppText color="light">{sampleText}</AppText>);
    const text = screen.getByText(sampleText);
    expect(text).toBeDefined();
    expect(text).toHaveStyle({
      color,
    });
  });

  it("Should render accent color variant", () => {
    const sampleText = "Code test";
    const color = theme.colors.colorAccent;
    render(<AppText color="accent">{sampleText}</AppText>);
    const text = screen.getByText(sampleText);
    expect(text).toBeDefined();
    expect(text).toHaveStyle({
      color,
    });
  });

  it("Should render custom color color variant", () => {
    const sampleText = "Code test";
    const color = "#FB7185";
    render(<AppText color={color}>{sampleText}</AppText>);
    const text = screen.getByText(sampleText);
    expect(text).toBeDefined();
    expect(text).toHaveStyle({
      color,
    });
  });

  it("Should render small font size variant", () => {
    const sampleText = "Code test";
    const fontSize = theme.fontSize.small;
    render(<AppText size="s">{sampleText}</AppText>);
    const text = screen.getByText(sampleText);
    expect(text).toBeDefined();
    expect(text).toHaveStyle({
      fontSize,
    });
  });

  it("Should render medium font size variant", () => {
    const sampleText = "Code test";
    const fontSize = theme.fontSize.medium;
    render(<AppText size="m">{sampleText}</AppText>);
    const text = screen.getByText(sampleText);
    expect(text).toBeDefined();
    expect(text).toHaveStyle({
      fontSize,
    });
  });

  it("Should render large font size variant", () => {
    const sampleText = "Code test";
    const fontSize = theme.fontSize.large;
    render(<AppText size="l">{sampleText}</AppText>);
    const text = screen.getByText(sampleText);
    expect(text).toBeDefined();
    expect(text).toHaveStyle({
      fontSize,
    });
  });

  it("Should render regular font variant", () => {
    const sampleText = "Code test";
    const fontFamily = theme.fontFamily.regular;
    render(<AppText type="regular">{sampleText}</AppText>);
    const text = screen.getByText(sampleText);
    expect(text).toBeDefined();
    expect(text).toHaveStyle({
      fontFamily,
    });
  });

  it("Should render semi bold font variant", () => {
    const sampleText = "Code test";
    const fontFamily = theme.fontFamily.semiBold;
    render(<AppText type="semiBold">{sampleText}</AppText>);
    const text = screen.getByText(sampleText);
    expect(text).toBeDefined();
    expect(text).toHaveStyle({
      fontFamily,
    });
  });

  it("Should render bold font variant", () => {
    const sampleText = "Code test";
    const fontFamily = theme.fontFamily.bold;
    render(<AppText type="bold">{sampleText}</AppText>);
    const text = screen.getByText(sampleText);
    expect(text).toBeDefined();
    expect(text).toHaveStyle({
      fontFamily,
    });
  });

  it("Should render bold large light variant", () => {
    const sampleText = "Code test";
    const fontFamily = theme.fontFamily.bold;
    const color = theme.colors.colorLight;
    const fontSize = theme.fontSize.large;
    render(
      <AppText type="bold" size="l" color="light">
        {sampleText}
      </AppText>,
    );
    const text = screen.getByText(sampleText);
    expect(text).toBeDefined();
    expect(text).toHaveStyle({
      fontFamily,
      color,
      fontSize,
    });
  });
});
