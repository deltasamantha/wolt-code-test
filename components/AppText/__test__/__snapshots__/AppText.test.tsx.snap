// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppText Should render AppText properly 1`] = `
<Text
  style={
    [
      {
        "color": "#202125",
        "fontFamily": "Roboto_400Regular",
        "fontSize": 12,
        "lineHeight": 16,
      },
      undefined,
    ]
  }
>
  Code test
</Text>
`;
